ocaml/parsing/parser.mly

```diff
@@ -130,19 +130,24 @@ let neg_string f =
 let mkuminus ~oploc name arg =
   match name, arg.pexp_desc with
   | "-", Pexp_constant(Pconst_integer (n,m)) ->
-      Pexp_constant(Pconst_integer(neg_string n,m)), arg.pexp_attributes
+      Pexp_constant(Pconst_integer(neg_string n,m)),
```

Comment:
I'm wondering if we can revert the changes to the file from this commit (afefad0b9019c02fec1aa4f224df82c5cb77d7cb) and instead make Jane Syntax not complain if there are no locations in the stack.

This diff in this file is of the category "tech debt we'll never pay off", so I'm inclined to not take it out to begin with. And, the diff doesn't seem clearly worth it to me: If someone writes `local_ 3`, it doesn't seem that bad to report the location as `local_ 3` and not `3`.

Reply:
I don't understand "This diff in this file is of the category 'tech debt we'll never pay off"=', so I'm inclined to not take it out to begin with.", it seems to contradict your first paragraph.

I'd like to avoid weird issues with locations changing, so I'd prefer to save it; however, I could minimize the diff.  Would that assuage your concerns?

Reply:
The paragraphs are not meant to contradict. I see these as the two opposing alternatives:
  * Keep the code as it is in your latest commit. This has better handling of locations.
  * Revert `afefad0`, and teach Jane Syntax to not error out when the location stack is empty. (Instead, it just won't change any locations.)

where the second bullet is what I was trying to express with my comment above. My point with "it's tech debt we'll never pay off" is that this diff will never go away as we'll never upstream it, which makes it more costly tech debt to take out.

As a further point in favor of the second bullet, I believe the diff `afefad0` in `parser.mly` only affects location handling for unboxed floats/ints &mdash; which is unaffected by this location-handling code anyway! (Unlike `local_ e`, unboxed floats don't involve handing the AST with restored location back to the caller.)

Reply:
As discussed, I minimized the diff